src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -369,6 +380,242 @@ void CascadiaSettings::_LoadDynamicProfiles()
+// - The set of ignored namespaces
+void CascadiaSettings::_ApplyJsonStubsHelper(const std::wstring_view directory, const std::unordered_set<std::wstring>& ignoredNamespaces)
+{
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };
+
+    // The json files should be within subdirectories where the subdirectory name is the app name
+    for (auto& protoExtFolder : std::filesystem::directory_iterator(root))
+    {
+        auto folderPath = protoExtFolder.path().generic_string();
+
+        // We only want the parent folder name as the source (not the full path)
+        auto source = til::u8u16(folderPath.substr(folderPath.find_last_of("/") + 1));
+
+        // Only apply the stubs if the parent folder name is not in ignored namespaces
+        // (also make sure this is a directory for sanity)
+        if (std::filesystem::is_directory(protoExtFolder) && ignoredNamespaces.find(source) == ignoredNamespaces.end())
+        {
+            const auto jsonFiles = _AccumulateJsonFilesInDirectory(protoExtFolder.path().c_str());
+            _AddOrModifyProfiles(jsonFiles, winrt::hstring{ source });
+        }
+    }
+}
+
+// Method Description:
+// - Finds all the json files within the given directory
+// Arguments:
+// - directory: the directory to search
+// Return Value:
+// - A set containing all the found file data
+std::unordered_set<std::string> CascadiaSettings::_AccumulateJsonFilesInDirectory(const std::wstring_view directory)
+{
+    std::unordered_set<std::string> jsonFiles;
+
+    // Expand out environment strings like %LOCALAPPDATA% and %ProgramData%
+    const std::filesystem::path root{ wil::ExpandEnvironmentStringsW<std::wstring>(directory.data()) };
+
+    for (auto& protoExt : std::filesystem::directory_iterator(root))
+    {
+        if (protoExt.path().extension() == jsonExtension)
+        {
+            wil::unique_hfile hFile{ CreateFileW(protoExt.path().c_str(),
+                                                 GENERIC_READ,
+                                                 FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                                 nullptr,
+                                                 OPEN_EXISTING,
+                                                 FILE_ATTRIBUTE_NORMAL,
+                                                 nullptr) };
+
+            if (hFile)
+            {
+                const auto fileData = _ReadFile(hFile.get()).value();
+                jsonFiles.emplace(fileData);
+            }
+        }
+    }
+    return jsonFiles;
+}
+
+// Method Description:
+// - Given a set of json files, uses them to modify existing profiles,
+//   create new profiles, and create new colour schemes
+// Arguments:
+// - stubs: the set of json files
+// - source: the location the files came from
+void CascadiaSettings::_AddOrModifyProfiles(const std::unordered_set<std::string> files, const winrt::hstring source)
+{
+    for (const auto file : files)

```

Comment:
```suggestion
    for (const auto& file : files)

```

Use the ref if you can help it to avoid pulling a copy of the string out of the set in loops like this.

Check elsewhere in this PR to see if you might have managed a similar thing.